pkgs/applications/misc/onboard/default.nix

```diff
@@ -0,0 +1,144 @@
+{ config
+, fetchurl
+, atspiSupport ? true, at_spi2_core ? null
+, gtk3
+, gnome3
+, libcanberra_gtk3
+, libudev
+, hunspell
+, isocodes
+, pkgconfig
+, xorg
+, libxkbcommon
+, python3
+, stdenv
+, bash
+, intltool
+, glib
+, glibcLocales
+, gobjectIntrospection
+, gsettings_desktop_schemas
+, procps
+, wrapGAppsHook
+, yelp
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  name = "onboard-${version}";
+  majorVersion = "1.4";
+  version = "${majorVersion}.1";
+  src = fetchurl {
+    url = "https://launchpad.net/onboard/${majorVersion}/${version}/+download/${name}.tar.gz";
+    sha256 = "01cae1ac5b1ef1ab985bd2d2d79ded6fc99ee04b1535cc1bb191e43a231a3865";
+  };
+
+  # Uncomment for tests
+  #LC_ALL = "en_US.UTF-8";
+  doCheck = false;
+
+  propagatedBuildInputs = [
+    python3
+    python3.pkgs.pycairo
+    python3.pkgs.dbus-python
+    python3.pkgs.pygobject3
+    python3.pkgs.systemd
+    python3.pkgs.distutils_extra
+    python3.pkgs.pyatspi
+    glib
+  ];
+
+  buildInputs = [
+    #python3.pkgs.nose # Needed for testing
+    gtk3
+    libcanberra_gtk3
+    libudev
+    bash
+    hunspell
+    isocodes
+    pkgconfig
+    xorg.libXtst
+    xorg.libxkbfile
+    libxkbcommon
+    intltool
+    glibcLocales
+    gsettings_desktop_schemas
+    gnome3.dconf
+    wrapGAppsHook
+  ] ++ stdenv.lib.optional atspiSupport at_spi2_core;
+
+  propagatedUserEnvPkgs = [
+    gnome3.dconf
+  ];
+
+  preBuild = ''
+    # Unnecessary file, has been removed upstream
+    # https://github.com/NixOS/nixpkgs/pull/24986#issuecomment-296114062
+    rm -r Onboard/pypredict/attic
+
+    substituteInPlace  ./scripts/sokSettings.py \
+      --replace "#!/usr/bin/python3" "" \
+      --replace "PYTHON_EXECUTABLE," "\"$out/bin/onboard-settings\"" \
+      --replace '"-cfrom Onboard.settings import Settings\ns = Settings(False)"' ""
+
+    chmod -x ./scripts/sokSettings.py
+
+    patchShebangs .
+
+    substituteInPlace  ./Onboard/LanguageSupport.py \
+      --replace "/usr/share/xml/iso-codes" "${isocodes}/share/xml/iso-codes" \
+      --replace "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./Onboard/Indicator.py \
+      --replace   "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./gnome/Onboard_Indicator@onboard.org/extension.js \
+      --replace "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./Onboard/SpellChecker.py \
+      --replace "/usr/share/hunspell" ${hunspell}/bin/hunspell \
```

Comment:
Also, tests require `(hunspellWithDicts [hunspellDicts.en-us])`.

Reply:
So just put something like the following in ```checkInputs```? 

```
  checkInputs = [
    python3.pkgs.nose
    (aspellWithDicts (dicts: with dicts; [ en ]))
    (hunspellWithDicts [hunspellDicts.en-us])
  ];
```

Do I still just use ```aspell``` and ```hunspell``` as requirements in the function parameter list?  Or will I need to add ```aspellWithDicts``` and ```hunspellWithDicts```.

Reply:
Yes, for the `nose` and `aspellWithDicts`.

What `buildInputs` does is that it adds the packages to `PATH` and other variables like `NIX_CFLAGS_COMPILE`. Since the source code only contains the absolute path to `hunspell` binary, adding the `hunspellWithDicts` to inputs will not have any effect.

Adding the `aspellWithDicts` to inputs, on the other hand, will be useful, since it is called with just the program name.

Also please note that unless the `PATH` is copied e.g. to a wrapper, or binary in the path fixed during build using, for example, using `which` command – and python packages usually do no such things – it will not have any effect on the resulting package.

----

Yes, you will need to have any derivations you use injected through a function parameter.

Reply:
So what should I do regarding ```hunspellWithDicts``` then if it will not have any effect?

Reply:
Replace the `${hunspell}` in the substitution by `${hunspellWithDicts [hunspellDicts.en-us]}`. Or better, define it beforehand:

```nix
let
  hunspell = hunspellWithDicts [hunspellDicts.en-us];
in python3.pkgs.buildPythonApplication rec {
...
}
```